test/boost/sstable_tablet_streaming.cc

```diff
@@ -0,0 +1,364 @@
+            auto ssts = make_sstables_with_ranges(env,
+                                                  {
+                                                      {50, 100},  // touches start -> partial
+                                                      {100, 120}, // starts at start -> fully contained
+                                                      {180, 200}, // ends at end   -> fully contained
+                                                      {200, 220}, // touches end   -> partial
+                                                  });
+            auto res = get_sstables_for_tablets(ssts, std::move(collection)).get();
+            REQUIRE_WITH_CONTEXT(res[0]._sstables_fully_contained, 2);
+            REQUIRE_WITH_CONTEXT(res[0]._sstables_partially_contained, 2);
+        }
+
+        // 10) Large SSTable set where early break should occur
+        {
+            auto collection = get_tablet_sstable_collection({dht::token{1000}, dht::token{2000}});
+            auto ssts = make_sstables_with_ranges(env,
+                                                  {
+                                                      {100, 200},
+                                                      {300, 400},
+                                                      {900, 950},
+                                                      {1001, 1100}, // fully contained
+                                                      {1500, 1600}, // fully contained
+                                                      {2101, 2200}, // entirely after -> should trigger early break in ascending scan
+                                                      {1999, 2100}, // overlap, partially contained
+                                                      {3000, 3100},
+                                                  });
+            auto res = get_sstables_for_tablets(ssts, std::move(collection)).get();
+            REQUIRE_WITH_CONTEXT(res[0]._sstables_fully_contained, 2);
+            REQUIRE_WITH_CONTEXT(res[0]._sstables_partially_contained, 1);
+        }
+
+        // 10) https://github.com/scylladb/scylladb/pull/26980 example, tested
+        {
+            auto collection = get_tablet_sstable_collection({dht::token{4}, dht::token{5}});
+            auto ssts = make_sstables_with_ranges(env,
+                                                  {
+                                                      {0, 5},
+                                                      {0, 3},
+                                                      {2, 5},
+                                                  });
+            auto res = get_sstables_for_tablets(ssts, std::move(collection)).get();
+            // None fully contained; three partial overlaps
+            REQUIRE_WITH_CONTEXT(res[0]._sstables_fully_contained, 0);
+            REQUIRE_WITH_CONTEXT(res[0]._sstables_partially_contained, 2);
+        }
+    });
+}
+
+SEASTAR_TEST_CASE(test_streaming_ranges_distribution_in_tablets) {
```

Comment:
Other than the ranges values themselves, Is the only difference from test_streaming_ranges_distribution in the number of tablets in the collection?